### PR TITLE
added flutter_lint

### PR DESCRIPTION
**Pull request by Yugue**
*Added flutter_lint*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: release_dashboard/analysis_options.yaml [hunk 1]
```diff
@@ -1,6 +1,8 @@
+include: package:flutter_lints/flutter.yaml
+
 linter:
   rules:
     avoid_catches_without_on_clauses: true
```
Comment (CaseyHillers) [inline release_dashboard/analysis_options.yaml:5]: Are these rules in flutter.yaml? Can we remove them?

File: release_dashboard/analysis_options.yaml [hunk 2]
```diff
@@ -1,6 +1,8 @@
...
     prefer_relative_imports: true
     unawaited_futures: true
-    ignore_todo: true
+    ignore_todo: true
```
Comment (CaseyHillers) [inline release_dashboard/analysis_options.yaml:8]: We should this to `flutter_style_todos`

File: release_dashboard/analysis_options.yaml (hunk 1 above)
Comment (Yugue) [reply release_dashboard/analysis_options.yaml:5]: I just saw another analysis_option file in the parent directory of `flutter/flutter` that the old release_dashboard was calling: https://github.com/flutter/flutter/blob/master/analysis_options.yaml

Should we add that file to the parent directory of Cocoon?

Comment (CaseyHillers) [reply release_dashboard/analysis_options.yaml:5]: No, flutter/flutter has a different style guide than what is recommended for flutter apps

File: release_dashboard/analysis_options.yaml (hunk 2 above)
Comment (CaseyHillers) [reply release_dashboard/analysis_options.yaml:8]: Switch this lint to `flutter_style_todos`

File: release_dashboard/analysis_options.yaml (hunk 1 above)
Comment (Yugue) [reply release_dashboard/analysis_options.yaml:5]: `avoid_catches_without_on_clauses: true` is used in the release_dashboard of 'flutter/flutter`, same for all the other ones, except `flutter_style_todos` which is not present in `flutter/flutter`.

Comment (Yugue) [reply release_dashboard/analysis_options.yaml:5]: which ones you are suggesting to remove?

Comment (Yugue) [reply release_dashboard/analysis_options.yaml:5]: This is what release dashboard in `flutter/flutter` uses:
```
    avoid_print: false
    only_throw_errors: false
    avoid_catching_errors: false
    avoid_catches_without_on_clauses: true
    prefer_relative_imports: true
    unawaited_futures: true
    ```

Comment (Yugue) [reply release_dashboard/analysis_options.yaml:5]: Oh I see what you mean. I just verified, none of the rules here are present in the `recommended` options.

Comment (Yugue) [reply release_dashboard/analysis_options.yaml]: I kept `flutter_style_todos` and `prefer_relative_imports: true`

File: release_dashboard/analysis_options.yaml [hunk 3]
```diff
@@ -1,6 +1,6 @@
+include: package:flutter_lints/flutter.yaml
+
 linter:
   rules:
-    avoid_catches_without_on_clauses: true
     prefer_relative_imports: true
-    unawaited_futures: true
-    ignore_todo: true
+    flutter_style_todos: true
```
Comment (CaseyHillers) [inline release_dashboard/analysis_options.yaml:6]:
```suggestion
    flutter_style_todos: true

```

Files should end with a newline

**Comment by CaseyHillers** [comment]
The checks on this are failing as the lints need to be fixed in the code now (running flutter analyze should surface them locally)

**Comment by CaseyHillers** [comment]
Make your to rebase your PRs once this lands. Otherwise, they may accidentally break the tree

**Pull request merged**